src/core/portal/dom-portal-host.ts

```diff
@@ -0,0 +1,56 @@
+import {DynamicComponentLoader, AppViewManager, ComponentRef} from 'angular2/core';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {BasePortalHost, ComponentPortal, TemplatePortal} from './portal';
+
+
+/**
+ * A PortalHost for attaching portals to an arbitrary DOM element outside of the Angular
+ * application context.
+ *
+ * This is the only part of the portal core that directly touches the DOM.
+ */
+export class DomPortalHost extends BasePortalHost {
+  constructor(
+      private _hostDomElement: Element,
```

Comment:
I went down that road with Tobias, and the advice from him is to just do the DOM stuff where you need it. Eventually Angular should provide a common way to do marshalling when web-workers are a thing. As it is now, an instance of `Renderer` is always tied to a directive and is meant for manipulating its view.